x-pack/test/alerting_api_integration/security_and_spaces/group1/tests/alerting/retain_api_key.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { UserAtSpaceScenarios } from '../../../scenarios';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import { AlertUtils, getUrlPrefix, getTestRuleData, ObjectRemover } from '../../../../common/lib';
+
+// eslint-disable-next-line import/no-default-export
+export default function retainAPIKeyTests({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+
+  describe('retain api key', () => {
+    const objectRemover = new ObjectRemover(supertest);
+
+    after(() => objectRemover.removeAll());
+
+    for (const scenario of UserAtSpaceScenarios) {
+      const { user, space } = scenario;
+      const alertUtils = new AlertUtils({ user, space, supertestWithoutAuth });
+
+      describe(scenario.id, () => {
+        it('should retain the api key when a rule is disabled and then enabled', async () => {
+          const { body: createdConnector } = await supertest
+            .post(`${getUrlPrefix(space.id)}/api/actions/connector`)
+            .set('kbn-xsrf', 'foo')
+            .send({
+              name: 'MY action',
+              connector_type_id: 'test.noop',
+              config: {},
+              secrets: {},
+            })
+            .expect(200);
```

Comment:
i was not sure about it and checked the other tests and it was not removed there too.
And thought that `after(() => objectRemover.removeAll());` does that for all the objects.
So i left it :) 
Adding now...

Reply:
ya, I think we've gotten lazy with connectors, not always setting them up to be deleted when we're done with tests. 
 And maybe someday that will bite us :-).  Rules we really have to remove, otherwise they continue to run and things can get confusing.  

The `objectRemover.removeAll()` does remove all the objects - that you tell it to remove - not everything.